lib/internal/errors.js

```diff
@@ -120,8 +112,13 @@ const prepareStackTrace = (globalThis, error, trace) => {
   //     at function (file)
   //     at file
   let errorString;
-  if (kIsNodeError in error) {
-    errorString = `${error.name} [${error.code}]: ${error.message}`;
+  // Do not use primordials here: we intercept user code here.
+  if (typeof error.toString === 'function' &&
+      error.toString !== MainContextObjectToString) {
+    errorString = error.toString();
+    if (errorString === '[Object object]') {
```

Comment:
Wouldn't it be better to have another `errorToString !== MainContextObjectPrototypeToString`?

Reply:
You mean to have a variable with the value assigned instead of using a "magic string"? I have no strong opinion either way.